rust/abacus-core/src/db/abacus_db.rs

```diff
@@ -231,4 +232,24 @@ impl AbacusDB {
         let value: Option<u32> = self.retrieve_keyed_decodable(LEAF_PROCESS_STATUS, &leaf_index)?;
         Ok(value.map(|x| x == 1))
     }
+
+    /// Update the total gas payment for a leaf index.
+    /// The provided gas payment amount is added to any existing stored gas payment in the DB.
```

Comment:
Is this going to be workable? I think one aspect of the current indexing scheme is that it is idempotent. This obviously isn't and I'm not sure that the way Abacus Works or other folks might be operating this, makes this safe always? Since performance seems much less of an issue, I would maybe suggest to store GasPayment events more normalized